File: src/main/java/dev/interfiber/karpet/server/utils/Probability.java
package dev.interfiber.karpet.server.utils;


import java.util.Random;

public class Probability {
    public static boolean Calculate(double Percent){
        Random r = new Random();
        int rng = r.nextInt(100);
        if (rng < Percent){
            return true;
        } else {
            return false;
        }
    }
}

File: src/main/kotlin/dev/interfiber/karpet/server/utils/Probability.kt
package dev.interfiber.karpet.server.utils

import java.util.Random

/**
 * Calculate probability
 * @author Interfiber
 */
object Probability {
    /**
     * Calculate a probability based on a percent
     * @author Interfiber
     */
    fun calculate(Percent: Double): Boolean {
        val r = Random()
        val rng = r.nextInt(100)
        return rng < Percent
    }
}
